### PR TITLE
Adds support for Attachments and VersionName

### LINKED ISSUE #21: Add support for Document Attachments
### Problem statement : 
Cannot create a SSM Document that can be used with AWS Distributor since attachments cannot be added to the definition. 

See an example of what the SSM Document looks like for a Distributor package using Cloudformation: 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-document.html#aws-resource-ssm-document--examples--Create_a__Distributor_package

### Solution: 
* Update lib to allow for an optional Attachments property at the Document level
   * Also will likely want to create AttachmentSource child object https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_AttachmentsSource.html
* Update lambda when create/update is called to pass the Attachments if they are defined

Happy to create a PR to implement this.

### PR DESCRIPTION
**Pull request by zendern**
Fixes #21

Also as the title states add VersionName since those are nice to haves when creating Distributor packages. 

Testing: 

Deployed and all existing Documents are created successfully. Attached are some screenshots of the new Distributor package that uses both Attachments and VersionName. 

## Post initial version creation. 
You can see the VersionName and the fact that it is in Active status (see comments in the README about granting access to make that work)
Screen Shot 2022-02-04 at 3 40 58 PM

## Post update with another version
This is version 2. It contains a pointer to a different package as well as a different VersionName. 
Screen Shot 2022-02-04 at 3 42 09 PM
Screen Shot 2022-02-04 at 3 42 18 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,6 +137,80 @@ export class TestStack extends cdk.Stack {
...
+     * If access is not granted to the role that created the document you may see 
+     * an error like the following : 
+     * 
+     * ```
+     * Permanent download error: Source URL 's3://cdk-ssm-document-storage/v1/package.zip' reported:
+     * Access Denied (Service: Amazon S3; Status Code: 403; 
+     * Error Code: AccessDenied; Request  *ID:DES1XEHZTJ9R; S3 Extended Request ID: 
+     * A+u8sTGQ6bZpAwl2eXDLq4KTkoeYyQR2XEV+I=; Proxy: null)
+     * ```
+     */
```
Comment (zendern) [inline README.md:193]: Here is the comment regarding the Active status on a Distributor package. Basically the creator of the document must have access to the S3 bucket that the package lives in.

File: lambda/index.ts [hunk 2]
```diff
@@ -157,6 +165,36 @@ function updateDocumentAddTags(event: Event): Promise<Event> {
...
+function toPascalCase(string) {
+  return `${string}`
+    .replace(new RegExp(/[-_]+/, 'g'), ' ')
+    .replace(new RegExp(/[^\w\s]/, 'g'), '')
+    .replace(
+      new RegExp(/\s+(.)(\w*)/, 'g'),
+      ($1, $2, $3) => `${$2.toUpperCase() + $3.toLowerCase()}`
+    )
+    .replace(new RegExp(/\w/), s => s.toUpperCase());
+}
```
Comment (zendern) [inline lambda/index.ts:196]: Nested objects are not pascal-ized when using `pascalCaseProperties: true,` on the CustomResource. So doing that work here. 

Was torn on if this should be before the CustomResource is called or in the lambda (where it is currently) Open to suggestions.

File: test/lib/test-stack.ts [hunk 3]
```diff
@@ -61,6 +64,77 @@ export class TestStack extends cdk.Stack {
...
+    // );
+    // const docE = new Document(this, `SSM-Distribution-Package`, {
+    //   documentType: 'Package',
+    //   name: 'Test-Distributon-Package',
+    //   content: fs.readFileSync(file).toString(),
+    //   versionName: '2.0-Better-Than_Sliced_Bread',
+    //   attachments: [
+    //     { key: "SourceUrl", values: [`s3://${bucketName}/v2`] }
+    //   ]
+    // });
```
Comment (zendern) [inline test/lib/test-stack.ts:114]: As it states this is how I tested update. Comment out the `docE` and then uncomment the follow chunk that is commented out.

File: test/lib/test-stack.ts [hunk 4]
```diff
@@ -61,6 +64,77 @@ export class TestStack extends cdk.Stack {
...
+     */
+    docE.lambda.role?.attachInlinePolicy(
+      new iam.Policy(this, 'distributor-s3-bucket-read', {
+        statements: [
+          new iam.PolicyStatement({
+            effect: iam.Effect.ALLOW,
+            actions: ['s3:GetObject'],
+            resources: [
+              `arn:aws:s3:::${bucketName}`,
+              `arn:aws:s3:::${bucketName}/*`
```
Comment (fitzoh) [inline test/lib/test-stack.ts:129]: Is the `arn:aws:s3:::${bucketName}` necessary?

**Comment by udondan** [comment]
Looking good and working nicely! I'll just make some minor adjustments.

File: test/lib/test-stack.ts (hunk 4 above)
Comment (udondan) [reply test/lib/test-stack.ts:129]: I think it's not. GetObject should only require object resources.

**Comment by udondan** [comment]
The test is failing, because the PR doesn't have access to the repo secrets. Haven't found a way to grant the test access, when the PR was not initiated by myself... tested it locally instead :)

**Comment by udondan** [comment]
Thanks for your contribution @zendern! Version 3.1.0 is now published. 🚀

**Pull request merged**